pilot/pkg/proxy/envoy/proxy.go

```diff
@@ -140,9 +143,18 @@ func (e *envoy) Run(config interface{}, epoch int, abort <-chan error) error {
 	if err := cmd.Start(); err != nil {
 		return err
 	}
+	// record envoy process number
+	e.processNum.Inc()
 	done := make(chan error, 1)
 	go func() {
-		done <- cmd.Wait()
+		err := cmd.Wait()
+		e.processNum.Dec()
+		if err == nil {
+			if e.processNum.Load() == 0 {
```

Comment:
Does the pilot agent currently try to run multiple envoy processes at the same time?

Reply:
Only on SIGTERM to drain and cert rotation.

Reply:
Ok, makes sense.   @liamawhite do you want to take a pass as well?